src/main/java/ch/njol/skript/expressions/ExprPermissions.java

```diff
@@ -31,58 +40,117 @@
+})
 @Examples("set {_permissions::*} to all permissions of the player")
-@Since("2.2-dev33")
+@Since("2.2-dev33, INSERT VERSION (Changers)")
 public class ExprPermissions extends SimpleExpression<String> {
-	
+
 	static {
-		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY, "[(all [[of] the]|the)] permissions (from|of) %players%", "[(all [[of] the]|the)] %players%'[s] permissions");
+		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY,
+			"[all [[of] the]|the] permissions (from|of) %entities%",
+			"[all [of] the] %entities%'[s] permissions"
+		);
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<Player> players;
-	
-	@SuppressWarnings({"null", "unchecked"})
+
+	private Expression<Entity> entities;
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		players = (Expression<Player>) exprs[0];
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		entities = (Expression<Entity>) exprs[0];
 		return true;
 	}
 
 	@Override
 	@Nullable
-	protected String[] get(Event e) {
-		final Set<String> permissions = new HashSet<>();
-		for (Player player : players.getArray(e))
-			for (final PermissionAttachmentInfo permission : player.getEffectivePermissions())
-				permissions.add(permission.getPermission());
-		return permissions.toArray(new String[permissions.size()]);
+	protected String[] get(Event event) {
+		return entities.stream(event)
+				.flatMap(permissible -> permissible.getEffectivePermissions().stream())
+				.map(permission -> permission.getPermission())
+				.toArray(String[]::new);
 	}
-	
+
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.ADD || mode == ChangeMode.REMOVE || mode == ChangeMode.DELETE)
+			return CollectionUtils.array(String[].class);
+		return null;
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		List<String> permissions = new ArrayList<>();
+		if (mode != ChangeMode.DELETE) {
+			if (delta == null)
+				return;
+			for (Object permission : delta) {
+				if (permission == null)
+					continue;
+				permissions.add((String) permission);
+			}
+		}
+		if (mode == ChangeMode.DELETE || mode == ChangeMode.REMOVE) {
+			for (Entity entity : entities.getAll(event)) {
+				for (PermissionAttachmentInfo info : entity.getEffectivePermissions()) {
+					PermissionAttachment attachment = info.getAttachment();
+					if (attachment == null)
+						continue;
+					for (String permission : attachment.getPermissions().keySet()) {
+						if (mode == ChangeMode.DELETE) {
+							attachment.unsetPermission(permission);
+							continue;
+						} else if (permissions.contains(permission)) {
+							attachment.unsetPermission(permission);
+						}
+					}
```

Comment:
```suggestion
					if (mode == ChangeMode.DELETE) {
						for (String permission : attachment.getPermissions().keySet())
							attachment.unsetPermission(permission);
					} else {
						for (String permission : permissions)
							attachment.unsetPermission(permission);
					}
```
Could even reduce it down to
```suggestion
					for (String permission : mode == ChangeMode.DELETE ? attachment.getPermissions().keySet() : permissions)
						attachment.unsetPermission(permission);
```
(it's a bit long though for the loop line)

Reply:
Attachment is null if it's a default permission. There must be a null check.

Reply:
For the first suggest, you need to check if the permission being removed exists in their permission set. You're removing that condition check.